engines/router/missionctl/server/application.go

```diff
@@ -47,37 +48,51 @@ func Run() {
 	// Init Sentry, defer closing client
 	defer initSentryClient(cfg)()
 
-	// Init mission control
-	missionCtl, err := missionctl.NewMissionControl(
-		nil,
-		cfg.EnrichmentConfig,
-		cfg.RouterConfig,
-		cfg.EnsemblerConfig,
-		cfg.AppConfig,
-	)
-	if err != nil {
-		log.Glob().Panicf("Failed initializing Mission Control: %v", err)
-	}
-
-	// Register handlers
-	http.Handle("/v1/internal/", http.StripPrefix(
-		"/v1/internal",
-		handlers.NewInternalAPIHandler([]string{
-			cfg.EnsemblerConfig.Endpoint,
-			cfg.EnrichmentConfig.Endpoint,
-		}),
-	))
-	http.Handle("/v1/predict", sentry.Recoverer(handlers.NewHTTPHandler(missionCtl)))
-	http.Handle("/v1/batch_predict", sentry.Recoverer(handlers.NewBatchHTTPHandler(missionCtl)))
-	// Register metrics handler
-	if cfg.AppConfig.CustomMetrics {
-		http.Handle("/metrics", promhttp.Handler())
-	}
-
-	// Serve
-	log.Glob().Infof("listening at port %d", cfg.Port)
-	if err := http.ListenAndServe(cfg.ListenAddress(), http.DefaultServeMux); err != nil {
-		log.Glob().Errorf("Failed to start Turing Mission Control API: %s", err)
+	switch cfg.RouterConfig.Protocol {
+	case config.UPI:
+		// Init mission control
+		missionCtl, err := missionctl.NewMissionControlUPI(
+			cfg.RouterConfig.ConfigFile,
+			cfg.AppConfig.FiberDebugLog,
+		)
+		if err != nil {
+			log.Glob().Panicf("Failed initializing Mission Control: %v", err)
+		}
+		s := upi.NewUPIServer(missionCtl, cfg.Port)
```

Comment:
We'll need to expose http metrics server in addition to the UPI gRPC server

Reply:
added [cmux](https://github.com/soheilhy/cmux) to host both grpc and http on the same port. Seems to work (screenshots in description)